src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -99,18 +100,22 @@ private static int[] ComputeKMPFailureFunction(byte[] pattern)
         }
 
         // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
-        private static int KMPSearch(byte[] pattern, byte[] bytes)
+        private static unsafe int KMPSearch(byte[] pattern, MemoryMappedViewAccessor accessor)
         {
             int m = 0;
             int i = 0;
             int[] table = ComputeKMPFailureFunction(pattern);
 
-            while (m + i < bytes.Length)
+            byte* ptreMemoryMappedView = (byte*)0;
+            accessor.SafeMemoryMappedViewHandle.AcquirePointer(ref ptreMemoryMappedView);
+
+            while (m + i < accessor.Capacity)
             {
-                if (pattern[i] == bytes[m + i])
+                if (pattern[i] == *(ptreMemoryMappedView + m + i))
```

Comment:
You can keep indexer syntax here pointers *p[a] == *(p + a). I'd even keep the bytes name for the pointer.

Reply:
C# is nice.

Fixed, change to bytes[m + i]